### PR TITLE
chore: fix complex type (clippy)

### PR DESCRIPTION
**Pull request by ecklf**
This PR extracts complex return types into dedicated type aliases as part of fixing `clippy` lint warnings.

Resolves WEB-185

### CODE REVIEW

File: crates/turbo-tasks-memory/src/stats.rs [hunk 1]
```diff
@@ -9,10 +9,16 @@ use std::{
...
 use crate::{
+    scope::TaskScopeId,
     task::{Task, TaskStatsInfo},
     MemoryBackend,
 };
 
+pub type StatsReferenceType = (
+    Vec<(ReferenceType, TaskId)>,
+    Vec<(ReferenceType, TaskScopeId)>,
+);
```
Comment (alexkirsz) [inline crates/turbo-tasks-memory/src/stats.rs:20]: It's worth making this a struct.

File: crates/turbopack-ecmascript/src/analyzer/linker.rs [hunk 2]
```diff
@@ -16,6 +16,8 @@ type LinkCacheValue = (
     Vec<Vec<(HashSet<Id>, Vec<(HashSet<Id>, JsValue)>)>>,
 );
 
+type LinkCacheItem = Option<(bool, JsValue, (HashSet<Id>, HashSet<Id>))>;
```
Comment (alexkirsz) [inline crates/turbopack-ecmascript/src/analyzer/linker.rs:19]: It's worth making this a struct as well.

**Review by alexkirsz — CHANGES_REQUESTED** [review]

File: crates/turbo-tasks-memory/src/stats.rs (hunk 1 above)
Comment (ecklf) [reply crates/turbo-tasks-memory/src/stats.rs:20]: Done

File: crates/turbopack-ecmascript/src/analyzer/linker.rs (hunk 2 above)
Comment (ecklf) [reply crates/turbopack-ecmascript/src/analyzer/linker.rs]: Done

File: crates/turbo-tasks-memory/src/stats.rs [hunk 3]
```diff
@@ -9,10 +9,16 @@ use std::{
...
 use crate::{
+    scope::TaskScopeId,
     task::{Task, TaskStatsInfo},
     MemoryBackend,
 };
 
+pub struct StatsReferences(
+    pub Vec<(ReferenceType, TaskId)>,
+    pub Vec<(ReferenceType, TaskScopeId)>,
+);
```
Comment (sokra) [inline crates/turbo-tasks-memory/src/stats.rs:20]:
```suggestion
pub struct StatsReferences {
    pub tasks: Vec<(ReferenceType, TaskId)>,
    pub scopes: Vec<(ReferenceType, TaskScopeId)>,
};
```

**Review by sokra — CHANGES_REQUESTED** [review]

**Pull request merged**